src/core/diffusion/euler_grid.cc

```diff
@@ -313,4 +313,100 @@ void EulerGrid::DiffuseWithNeumann(real_t dt) {
   c1_.swap(c2_);
 }
 
+void EulerGrid::DiffuseWithPeriodic(real_t dt) {
+
+  const auto nx = resolution_;
+  const auto ny = resolution_;
+  const auto nz = resolution_;
+  const auto num_boxes = nx * ny * nz;
+
+  const real_t dx = box_length_;
+  const real_t d = 1 - dc_[0];
+
+  constexpr size_t YBF = 16;
+#pragma omp parallel for collapse(2)
+  for (size_t yy = 0; yy < ny; yy += YBF) {
+    for (size_t z = 0; z < nz; z++) {
+      size_t ymax = yy + YBF;
+      if (ymax >= ny) {
+        ymax = ny;
+      }
+      for (size_t y = yy; y < ymax; y++) {
+        size_t x{0};
+        size_t c{0};
+        size_t l{0};
+        size_t r{0};
+        size_t n{0};
+        size_t s{0};
+        size_t b{0};
+        size_t t{0};
+        c = x + y * nx + z * nx * ny;
+#pragma omp simd
+        for (x = 0; x < nx; x++) {
+          l = c - 1;
+          r = c + 1;
+          n = c - nx;
+          s = c + nx;
+          b = c - nx * ny;
+          t = c + nx * ny;
+
+          // Clamp to avoid out of bounds access. Clamped values are initialized
+          // to a wrong value but will be overwritten by the boundary condition
+          // evaluation. All other values are correct.
+          real_t left{c1_[Clamp(l, 0, num_boxes - 1)]};
```

Comment:
Can you merge the branch `macos-cpp17` into your and replace all occurrences of this `Clamp` function with std::clamp? Not sure where else it appears but probably a good time to get rid of it.

Reply:
I've merged the branch but the clamp function is unmodified in all the other diffusion functions, should I replace it there as well?

Reply:
@ryan-bournes Yes, I would do that to keep consistency across the codebase. Anyway, that's throwing an error and it looks like it has to do with type inference